distsql/stream.go

```diff
@@ -64,7 +68,10 @@ func (r *streamResult) Next(ctx context.Context, chk *chunk.Chunk) error {
 
 // readDataFromResponse read the data to result. Returns true means the resp is finished.
 func (r *streamResult) readDataFromResponse(ctx context.Context, resp kv.Response, result *tipb.Chunk) (bool, error) {
+	startTime := time.Now()
 	resultSubset, err := resp.Next(ctx)
+	// TODO: Add a label to distinguish between success or failure
+	metrics.DistSQLQueryHistgram.WithLabelValues(r.label, r.sqlType).Observe(time.Since(startTime).Seconds())
```

Comment:
should be deferred?

Reply:
We should only gauge the time task data transferring from TiKV but not plus the time for data transactions.